src/main/java/seedu/address/logic/commands/MarkMeetingCommand.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+import java.util.Iterator;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.meeting.Attendee;
+import seedu.address.model.meeting.Meeting;
+import seedu.address.model.meeting.MeetingStatus;
+import seedu.address.model.person.Person;
+
+/**
+ * Marks a meeting as complete.
+ */
+public class MarkMeetingCommand extends Command {
+
+    public static final String COMMAND_WORD = "mark";
+
+    public static final String MESSAGE_MARK_MEETING_SUCCESS = "Meeting marked as complete: %1$s";
+
+    public static final String MESSAGE_MEETING_ALREADY_COMPLETE = "Meeting has already been marked as complete.";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Marks the meeting identified by the index number used in the displayed meetings list as complete.\n"
+            + "Parameters: INDEX (must be a positive integer)\n" + "Example: " + COMMAND_WORD + "1";
+
+    private final Index targetIndex;
+
+    public MarkMeetingCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meeting> lastShownList = model.getFilteredMeetingList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
+        }
+
+        Meeting meetingToMark = lastShownList.get(targetIndex.getZeroBased());
+        Meeting updatedMeeting = markMeeting(meetingToMark);
+        model.setMeeting(meetingToMark, updatedMeeting);
+
+        Iterator<Attendee> attendeeIterator = meetingToMark.getAttendees().iterator();
+        while (attendeeIterator.hasNext()) {
+            Attendee attendee = attendeeIterator.next();
+            Person person = model.getPerson(attendee.getAttendeeName());
+            Person updatedPerson = updateLastContactedTime(person, meetingToMark.getEnd());
+            model.setPerson(person, updatedPerson);
+        }
```

Comment:
I think this can be off loaded to ModelManager to have better SLAP, it also increases coupling as this MarkMeetingCommand is now depended on Attendee too. 

But if there is no time, pls ignore and focus on more pressing work.